aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -387,9 +387,67 @@ _embedding_bag_cpu(const Tensor &weight, const Tensor &indices,
   }
 }
 
+Tensor _embedding_bag_sparse_backward_cpu_sum_fast(
+    const Tensor &grad_, const Tensor &indices, const Tensor &offsets, int64_t num_weights, int64_t mode, const Tensor& per_sample_weights) {
+
+  AT_ASSERT(mode == MODE_SUM);
+  AT_ASSERT((grad_.scalar_type() == kFloat)&& (grad_.stride(1) == 1));
+
+  auto indices_size0 = indices.size(0);
+  auto ddim = grad_.size(1);
+  Tensor index_grad = at::empty({indices_size0, ddim}, grad_.options());
+  auto gradout_data = index_grad.data_ptr<float>();
+
+  auto offsets_accessor = offsets.accessor<int64_t, 1>();
+  auto offset_numel = offsets.numel();
+
+  auto grad_data = grad_.data_ptr<float>();
+  auto grad_stride0 = grad_.stride(0);
+  float* scale_data = NULL;
+  auto scale_stride = 1;
+  if (per_sample_weights.defined()) {
+    AT_ASSERT(per_sample_weights.stride(0) == 1);
+    Tensor scale_weights = per_sample_weights.unsqueeze(1);
+    scale_data = scale_weights.data_ptr<float>();
+    scale_stride = per_sample_weights.stride(0);
+  }
+
+  at::parallel_for(0, offset_numel, 0, [&](int64_t start, int64_t end) {
```

Comment:
Why capture by reference?  It looks like everything here could be captured by value, and be strictly more efficient by removing an indirection.